cranelift/interpreter/src/step.rs

```diff
@@ -796,3 +812,78 @@ where
         }
     })
 }
+
+/// Converts a SIMD vector value into a Rust vector of i128 for processing.
+fn extractlanes<V>(x: &V, lane_type: types::Type) -> ValueResult<Vec<i128>>
```

Comment:
I think it might be slightly preferable here to use `SmallVec`, perhaps `SmallVec<[i128; 1]>` or similar (and typedef it so we can easily change it) -- while we don't expect the interpreter to be *fast*, it's still good to avoid an unnecessary heap allocation per execution of an instruction.

Reply:
Is this specifically for `extractlanes` or would this also apply to the input to `vectorizelanes`? Just want to check in case you wanted basically all the `Vec`s in this section of code changing, whilst I'm thinking that parameters aren't an issue (which I may be wrong about) and `Vec` parameters seem more general / flexible (accepting both `SmallVec` and `Vec`).

Reply:
I guess the advice applies generally. Specifically, this:

> `Vec` parameters seem more general / flexible (accepting both `SmallVec` and `Vec`)

is not quite right; `Vec` is a specific type that requires its contents to be heap-allocated. The more idiomatic Rust type for "any array of values in memory" would be a slice, i.e. `&[128]`, which one can get with `&vec[..]` or `&smallvec[..]`; but that only works for inbound args because it has no ownership. Returning a `SmallVec` is a technique we use elsewhere pretty successfully so I think it's' reasonable here too :-)

Reply:
Ah, I was basing that statement off testing where using a `Vec` parameter but passing a `SmallVec` was accepted by the compiler and built successfully, but the inverse (using a `SmallVec` parameter but passing a `Vec`) was blocked by the compiler. I wasn't sure about if that made the `SmallVec` behave like a `Vec` with regards to heap allocations, however, hence my question.
> Returning a SmallVec is a technique we use elsewhere pretty successfully

I understood that, and changed it to behave like that in 0089e15; I was more wondering about the input parameters to `vectorizelanes`. I'm gathering from your response though that the preference is for those to be a slice?